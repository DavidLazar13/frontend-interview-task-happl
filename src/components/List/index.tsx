import React, { useEffect, useState } from 'react'
import { Allowance } from '@/types/allowances'
import Card from '@/components/Card'

const List = () => {
  const [allowances, setAllowances] = useState<Allowance[]>([])

  useEffect(() => {
    const fetchData = async () => {
      const response = await fetch('/api/allowances')
      const { result } = await response.json()
      setAllowances(result)
    }

    fetchData()
  }, [])

  return (
    <div className='m-8'>
      <h2 className='text-2xl font-semibold leading-5 text-green-400'>Allowances</h2>
      <div className='grid grid-cols-1 sm:grid-cols-1 md:grid-cols-2 xl:grid-cols-3 gap-4 mt-7'>
        {allowances.map((allowance) => (
          <Card key={allowance.id} allowance={allowance}></Card>
        ))}
      </div>
    </div>
  )
}

export default List
