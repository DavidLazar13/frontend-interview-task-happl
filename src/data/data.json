[
  {
    "id": 1,
    "name": "Learning & Development",
    "renewal": "month",
    "currency": "£",
    "amount": "100",
    "spent": "20",
    "active": true,
    "type": "expense"
  },
  {
    "id": 2,
    "name": "Wellbeing",
    "renewal": "month",
    "currency": "£",
    "amount": "150",
    "spent": "140",
    "active": true,
    "type": "card"
  },
  {
    "id": 3,
    "name": "WFH",
    "renewal": "year",
    "currency": "£",
    "amount": "1000",
    "spent": "800",
    "active": true,
    "type": "card"
  },
  {
    "id": 4,
    "name": "Lunch",
    "renewal": "week",
    "currency": "£",
    "amount": "15",
    "spent": "0",
    "active": false,
    "type": "card"
  },
  {
    "id": 5,
    "name": "Team Day",
    "renewal": "year",
    "currency": "£",
    "amount": "200",
    "spent": "0",
    "active": false,
    "type": "card"
  },
  {
    "id": 6,
    "name": "Birthday",
    "renewal": "year",
    "currency": "£",
    "amount": "100",
    "spent": "100",
    "active": true,
    "type": "card"
  },
  {
    "id": 7,
    "name": "Software",
    "renewal": "month",
    "currency": "£",
    "amount": "30",
    "spent": "8.99",
    "active": true,
    "type": "card"
  },
  {
    "id": 8,
    "name": "Team Budget",
    "renewal": "year",
    "currency": "£",
    "amount": "400",
    "spent": "125",
    "active": true,
    "type": "card"
  }
]
