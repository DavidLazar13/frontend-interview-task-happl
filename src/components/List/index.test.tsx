import { render, waitFor } from '@testing-library/react'
import List from '@/components/List'
import '@testing-library/jest-dom'
import 'jest-fetch-mock'
import data from '@/data/data.json'

describe('List', () => {
  beforeEach(() => {
    fetchMock.mockResponse(JSON.stringify({ result: data }), { status: 200 })
  })

  it('renders as expected', async () => {
    const { container } = render(<List />)
    await waitFor(() => expect(fetchMock).toHaveBeenCalledTimes(1))
    await waitFor(() => expect(container).toMatchSnapshot())
  })

  it.skip('renders header as expected', async () => {})

  it.skip('renders list as expected', async () => {})

  it.skip('renders active allowance as expected', async () => {})

  it.skip('renders expense allowance as expected', async () => {})

  it.skip('renders card allowance as expected', async () => {})

  it.skip('renders inactive allowance as expected', async () => {})
})
