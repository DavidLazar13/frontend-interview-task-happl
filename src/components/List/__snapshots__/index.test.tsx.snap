// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List renders as expected 1`] = `
<div>
  <div
    class="m-8"
  >
    <h2
      class="text-2xl font-semibold leading-5 text-green-400"
    >
      Allowances
    </h2>
    <div
      class="grid grid-cols-1 sm:grid-cols-1 md:grid-cols-2 xl:grid-cols-3 gap-4 mt-7"
    />
  </div>
</div>
`;
